tests/all/stack_overflow.rs

```diff
@@ -9,35 +9,97 @@ fn host_always_has_some_stack() -> Result<()> {
+                ;; exit via wasm-to-native trampoline
+                (func $recursive1 (export "f1")
+                    call $host1
+                    call $recursive1)
+
+                ;; exit via wasm-to-array trampoline
+                (func $recursive2 (export "f2")
+                    call $host2
+                    call $recursive2)
+
+                ;; exit via a wasmtime-based libcall
+                (memory 1)
+                (func $recursive3 (export "f3")
+                    (drop (memory.grow (i32.const 00)))
+                    call $recursive3)
+
+                ;; exit via a cranelift-based libcall
+                (func $recursive4 (export "f4")
+                    (drop (call $f32_ceil (f32.const 0)))
+                    call $recursive4)
+                (func $f32_ceil (param f32) (result f32)
+                    (f32.ceil (local.get 0)))
             )
         "#,
     )?;
-    let func = Func::wrap(&mut store, test_host_stack);
-    let instance = Instance::new(&mut store, &module, &[func.into()])?;
-    let foo = instance.get_typed_func::<(), ()>(&mut store, "foo")?;
+    let host1 = Func::wrap(&mut store, test_host_stack);
+    let ty = FuncType::new(store.engine(), [], []);
+    let host2 = Func::new(&mut store, ty, |_, _, _| {
+        test_host_stack();
+        Ok(())
+    });
+    let instance = Instance::new(&mut store, &module, &[host1.into(), host2.into()])?;
+    let f1 = instance.get_typed_func::<(), ()>(&mut store, "f1")?;
+    let f2 = instance.get_typed_func::<(), ()>(&mut store, "f2")?;
+    let f3 = instance.get_typed_func::<(), ()>(&mut store, "f3")?;
+    let f4 = instance.get_typed_func::<(), ()>(&mut store, "f4")?;
 
     // Make sure that our function traps and the trap says that the call stack
     // has been exhausted.
-    let trap = foo.call(&mut store, ()).unwrap_err().downcast::<Trap>()?;
+    let hits1 = HITS.load(SeqCst);
+    let trap = f1.call(&mut store, ()).unwrap_err().downcast::<Trap>()?;
+    assert_eq!(trap, Trap::StackOverflow);
+    let hits2 = HITS.load(SeqCst);
+    let trap = f2.call(&mut store, ()).unwrap_err().downcast::<Trap>()?;
+    assert_eq!(trap, Trap::StackOverflow);
+    let hits3 = HITS.load(SeqCst);
+    let trap = f3.call(&mut store, ()).unwrap_err().downcast::<Trap>()?;
+    assert_eq!(trap, Trap::StackOverflow);
+    let hits4 = HITS.load(SeqCst);
+    let trap = f4.call(&mut store, ()).unwrap_err().downcast::<Trap>()?;
     assert_eq!(trap, Trap::StackOverflow);
+    let hits5 = HITS.load(SeqCst);
 
     // Additionally, however, and this is the crucial test, make sure that the
     // host function actually completed. If HITS is 1 then we entered but didn't
     // exit meaning we segfaulted while executing the host, yet still tried to
     // recover from it with longjmp.
-    assert_eq!(HITS.load(SeqCst), 0);
+    assert_eq!(hits1, 0);
+    assert_eq!(hits2, 0);
+    assert_eq!(hits3, 0);
+    assert_eq!(hits4, 0);
+    assert_eq!(hits5, 0);
```

Comment:
We never store to `HITS`; I assume we did at one point in time, but not anymore. Therefore, this isn't really testing what the comment says it is.

Reply:
Oh the store is [just below in `test_host_stack`](https://github.com/bytecodealliance/wasmtime/blob/9d1e2e0aa4e3d4153f8638085e0029cbc722c774/tests/all/stack_overflow.rs#L45)